### PR TITLE
fix: Improve TLS cert validation for LogPipeline

### PR DESCRIPTION
**Pull request by rakesh-garimella**
## Description

Changes proposed in this pull request (what was done and why):

- Improve tls handling for log pipeline to check corner cases
- Handle the case when there is bot tls cert and private key invalid. Then it should show cert as invalid
- If the secret is missing then it should have highest priority over tls cert validation
- Refactor the code 
Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/930

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [ ] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: internal/conditions/tls_cert_conditions.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kyma-project/telemetry-manager/internal/tlscert"
+)
+
+const twoWeeks = time.Hour * 24 * 7 * 2
+
+func EvaluateTLSCertCondition(certValidationResult tlscert.TLSCertValidationResult) (status metav1.ConditionStatus, reason, message string) {
+	if !certValidationResult.CertValid {
+		return metav1.ConditionFalse, ReasonTLSCertificateInvalid, fmt.Sprintf(getCommonMessage(ReasonTLSCertificateInvalid), certValidationResult.CertValidationMessage)
```
Comment (skhalash) [inline internal/conditions/tls_cert_conditions.go:16]: Please use a map indexer here

File: internal/reconciler/telemetry/log_components_checker.go [hunk 2]
```diff
@@ -109,15 +116,11 @@ func (l *logComponentsChecker) createMessageForReason(pipelines []telemetryv1alp
 	})
 }
 
-func determineFormattedTLSCertificateMessage(pipelines []telemetryv1alpha1.LogPipeline) string {
-
+func (l *logComponentsChecker) checkTLSCertificateMessage(pipelines []telemetryv1alpha1.LogPipeline) string {
```
Comment (skhalash) [inline internal/reconciler/telemetry/log_components_checker.go:119]:
```suggestion
func (l *logComponentsChecker) firstInvalidTLSPipelineMessage(pipelines []telemetryv1alpha1.LogPipeline) string {
```

File: internal/reconciler/logpipeline/reconciler.go [hunk 3]
```diff
@@ -319,76 +319,52 @@ func (r *Reconciler) calculateChecksum(ctx context.Context) (string, error) {
...
 // A pipeline is deployable if it is not being deleted, all secret references exist, and it doesn't have the legacy grafana-loki output defined.
-func getDeployableLogPipelines(ctx context.Context, allPipelines []telemetryv1alpha1.LogPipeline, client client.Client, certValidator TLSCertValidator) []telemetryv1alpha1.LogPipeline {
+func (r *Reconciler) getReconcilablePipelines(ctx context.Context, allPipelines []telemetryv1alpha1.LogPipeline) []telemetryv1alpha1.LogPipeline {
 	var deployablePipelines []telemetryv1alpha1.LogPipeline
 	for i := range allPipelines {
-		if !allPipelines[i].GetDeletionTimestamp().IsZero() {
-			continue
+		isReconcilable := r.isReconcilable(ctx, &allPipelines[i])
+		if isReconcilable {
+			deployablePipelines = append(deployablePipelines, allPipelines[i])
```
Comment (skhalash) [inline internal/reconciler/logpipeline/reconciler.go:329]:
```suggestion
			reconcilablePipelines = append(deployablePipelines, allPipelines[i])
```

Comment (rakesh-garimella) [reply internal/reconciler/logpipeline/reconciler.go]: fixed

File: internal/reconciler/telemetry/log_components_checker.go (hunk 2 above)
Comment (rakesh-garimella) [reply internal/reconciler/telemetry/log_components_checker.go]: fixed

File: internal/conditions/tls_cert_conditions.go (hunk 1 above)
Comment (rakesh-garimella) [reply internal/conditions/tls_cert_conditions.go]: fixed

**Review by skhalash — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**